pkgs/os-specific/linux/numatop/default.nix

```diff
@@ -16,12 +16,14 @@ stdenv.mkDerivation rec {
 
   patches = [
     (fetchpatch {
-      url = "https://github.com/intel/numatop/pull/54.patch";
+      # https://github.com/intel/numatop/54
+      url = "https://github.com/intel/numatop/commit/eab0ac5253c5843aa0f0ac36e2eec7612207711b.patch";
       sha256 = "sha256-TbMLv7TT9T8wE4uJ1a/AroyPPwrwL0eX5IBLsh9GTTM=";
       name = "fix-string-operations.patch";
```

Comment:
```
fix-string-operations.patch> 100  3188  100  3188    0     0   6866      0 --:--:-- --:--:-- --:--:--  6870
error: hash mismatch in fixed-output derivation '/nix/store/lsbp5i4z0d3a3lldasr68i5497qimg07-fix-string-operations.patch.drv':
         specified: sha256-TbMLv7TT9T8wE4uJ1a/AroyPPwrwL0eX5IBLsh9GTTM=
            got:    sha256-wDlb+SDFue+kNJAJFzG0sJd33/pBijI3PHym3cnd9jw=
```

diff

```diff
1,18c1,42
< --- a/common/os/os_util.c
< +++ b/common/os/os_util.c
< @@ -391,7 +391,7 @@
<  		return (B_FALSE);
<  	}
<  
< -	strncpy(scopy, str, len);
< +	strncpy(scopy, str, len + 1);
<  	scopy[len] = 0;
<  	cur = scopy;
<  
< @@ -447,7 +447,6 @@
<  		fclose(fp);
<  		return (B_FALSE);
<  	}
< -
<  	fclose(fp);
<  	return (str_int_extract(buf, arr, arr_size, num));
---
> From eab0ac5253c5843aa0f0ac36e2eec7612207711b Mon Sep 17 00:00:00 2001
> From: Colin Ian King <colin.i.king@gmail.com>
> Date: Fri, 1 Sep 2023 14:47:40 +0100
> Subject: [PATCH] Clean up some spelling mistakes in comment and manual
> 
> There are a few spelling mistakes found with codespell, fix these.
> 
> Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
> ---
>  common/disp.c | 2 +-
>  common/lwp.c  | 4 ++--
>  common/win.c  | 8 ++++----
>  numatop.8     | 2 +-
>  4 files changed, 8 insertions(+), 8 deletions(-)
> 
> diff --git a/common/disp.c b/common/disp.c
> index bb0bba2..ba2ce82 100644
> --- a/common/disp.c
> +++ b/common/disp.c
> @@ -637,7 +637,7 @@ cons_handler(void *arg __attribute__((unused)))
>  	win_fix_init();
>  
>  	/*
> -	 * Excute "home" command. It shows the NumaTop default page.
> +	 * Execute "home" command. It shows the NumaTop default page.
>  	 */
>  	disp_go_home();
>  
> diff --git a/common/lwp.c b/common/lwp.c
> index f2f5069..0b5abb4 100644
> --- a/common/lwp.c
> +++ b/common/lwp.c
> @@ -45,7 +45,7 @@
>  #include "include/os/os_util.h"
>  
>  /*
> - * Allocation and initialization for a 'track_lwp_t' struture.
> + * Allocation and initialization for a 'track_lwp_t' structure.
>   * A 'track_lwp_t' structure is allocated for each tracked thread.
>   */
>  static track_lwp_t *
> @@ -92,7 +92,7 @@ lwp_alloc(void)
20,79c44,103
< --- a/common/os/os_win.c
< +++ b/common/os/os_win.c
< @@ -117,6 +117,7 @@
<  	int i, j, k, l, cpuid_start;
<  	int *cpuid_arr;
<  	int ncpus;
< +	int s1_len = size;
<  	perf_cpu_t *cpus = node_cpus(node);
<  
<  	s1[0] = 0;
< @@ -140,8 +141,7 @@
<  	cpuid_start = cpuid_arr[0];
<  
<  	if (ncpus == 1) {
< -		(void) snprintf(s2, sizeof (s2), "%d", cpuid_start);
< -        	(void) strncat(s1, s2, strlen(s2));
< +		(void) snprintf(s1, size, "%d", cpuid_start);
<          	free(cpuid_arr);
<  		return;
<  	}
< @@ -152,6 +152,8 @@
<  	for (j = 1; j < ncpus; j++) {
<  		k++;
<  		if (cpuid_arr[j] != cpuid_start + l) {
< +			int s2_len = sizeof(s2);
< +
<  			if (k < ncpus) {
<  				if (l == 1) {
<  					(void) snprintf(s2, sizeof (s2), "%d ", cpuid_start);
< @@ -167,20 +169,27 @@
<  					(void) snprintf(s2, sizeof (s2), "%d-%d",
<  						cpuid_start, cpuid_start + l - 1);
<  				}
< +				s2_len -= strlen(s2);
<  
<  				(void) snprintf(s3, sizeof (s3), " %d",
<  					cpuid_arr[j]);
< -	        	  	(void) strncat(s2, s3, strlen(s3));
< +				s2_len -= strlen(s3);
< +				if (s2_len > 0)
< +					(void) strncat(s2, s3, s2_len);
<  			}
<  
< -          		(void) strncat(s1, s2, strlen(s2));
< +			s1_len -= strlen(s2);
< +			if (s1_len > 0)
< +				(void) strncat(s1, s2, s1_len);
<            		cpuid_start = cpuid_arr[j];
<             		l = 1;
<  		} else {
<  	        	if (k == ncpus) {
<          	    		(void) snprintf(s2, sizeof (s2), "%d-%d",
<                  			cpuid_start, cpuid_start + l);
< -         			(void) strncat(s1, s2, strlen(s2));
< +				s1_len -= strlen(s2);
< +				if (s1_len > 0)
< +					(void) strncat(s1, s2, s1_len);
<         			} else {
<              			l++;
<         			}
---
>  
>  /*
> - * Clean up the resource of 'track_lwp_t' struture. Before calling
> + * Clean up the resource of 'track_lwp_t' structure. Before calling
>   * lwp_free(), 'track_lwp_t' must be removed from 'lwp_list' in
>   * 'track_proc_t' structure.
>   */
> diff --git a/common/win.c b/common/win.c
> index d0a8f3b..f8fffb9 100644
> --- a/common/win.c
> +++ b/common/win.c
> @@ -456,7 +456,7 @@ topnproc_data_show(dyn_win_t *win)
>  	reg_refresh_nout(data_reg);
>  
>  	/*
> -	 * Dispaly hint message for window type
> +	 * Display hint message for window type
>  	 * "WIN_TYPE_TOPNPROC" and "WIN_TYPE_RAW_NUM"
>  	 */
>  	r = &dyn->hint;
> @@ -845,7 +845,7 @@ moniproc_data_show(dyn_win_t *win, boolean_t *note_out)
>  	proc_refcount_dec(proc);
>  
>  	/*
> -	 * Dispaly hint message for window type "WIN_TYPE_MONIPROC"
> +	 * Display hint message for window type "WIN_TYPE_MONIPROC"
>  	 */
>  	r = &dyn->hint;
>  	reg_erase(r);
> @@ -990,7 +990,7 @@ monilwp_data_show(dyn_win_t *win, boolean_t *note_out)
>  	proc_refcount_dec(proc);
>  
>  	/*
> -	 * Dispaly hint message for window type "WIN_TYPE_MONILWP"
> +	 * Display hint message for window type "WIN_TYPE_MONILWP"
>  	 */
>  	r = &dyn->hint;
>  	reg_erase(r);
> @@ -1634,7 +1634,7 @@ nodeoverview_data_show(dyn_win_t *win, boolean_t *note_out)
>  	reg_refresh_nout(r);
>  
>  	/*
> -	 * Dispaly hint message for window type "WIN_TYPE_NODE_OVERVIEW"
> +	 * Display hint message for window type "WIN_TYPE_NODE_OVERVIEW"
>  	 */
>  	r = &dyn->hint;
>  	reg_erase(r);
> diff --git a/numatop.8 b/numatop.8
> index 9eb983c..b82eaa4 100644
> --- a/numatop.8
> +++ b/numatop.8
> @@ -439,7 +439,7 @@ Specifies the level of logging in the log file. Valid values are:
>  -f log_file    
>  .br
>  Specifies the log file where output will be written. If the log file is
> -not writable, the tool will prompt "Cannot open '<file name>' for writting.".
> +not writable, the tool will prompt "Cannot open '<file name>' for writing.".
>  .PP    
>  -d dump_file
>  .br
```

Looks like the relevant changes were force-pushed out
A few commits can be recovered
https://github.com/intel/numatop/commit/ff46743eb8e31a4d0c2803fc09dc90912e9c990a
https://github.com/intel/numatop/commit/c1001fd926c24eae2d40729492e07270ce133b72

Let's vendor the patch, if it's still necessary

CC @dtzWill 

```patch
--- a/common/os/os_util.c
+++ b/common/os/os_util.c
@@ -391,7 +391,7 @@
 		return (B_FALSE);
 	}
 
-	strncpy(scopy, str, len);
+	strncpy(scopy, str, len + 1);
 	scopy[len] = 0;
 	cur = scopy;
 
@@ -447,7 +447,6 @@
 		fclose(fp);
 		return (B_FALSE);
 	}
-
 	fclose(fp);
 	return (str_int_extract(buf, arr, arr_size, num));
 }
--- a/common/os/os_win.c
+++ b/common/os/os_win.c
@@ -117,6 +117,7 @@
 	int i, j, k, l, cpuid_start;
 	int *cpuid_arr;
 	int ncpus;
+	int s1_len = size;
 	perf_cpu_t *cpus = node_cpus(node);
 
 	s1[0] = 0;
@@ -140,8 +141,7 @@
 	cpuid_start = cpuid_arr[0];
 
 	if (ncpus == 1) {
-		(void) snprintf(s2, sizeof (s2), "%d", cpuid_start);
-        	(void) strncat(s1, s2, strlen(s2));
+		(void) snprintf(s1, size, "%d", cpuid_start);
         	free(cpuid_arr);
 		return;
 	}
@@ -152,6 +152,8 @@
 	for (j = 1; j < ncpus; j++) {
 		k++;
 		if (cpuid_arr[j] != cpuid_start + l) {
+			int s2_len = sizeof(s2);
+
 			if (k < ncpus) {
 				if (l == 1) {
 					(void) snprintf(s2, sizeof (s2), "%d ", cpuid_start);
@@ -167,20 +169,27 @@
 					(void) snprintf(s2, sizeof (s2), "%d-%d",
 						cpuid_start, cpuid_start + l - 1);
 				}
+				s2_len -= strlen(s2);
 
 				(void) snprintf(s3, sizeof (s3), " %d",
 					cpuid_arr[j]);
-	        	  	(void) strncat(s2, s3, strlen(s3));
+				s2_len -= strlen(s3);
+				if (s2_len > 0)
+					(void) strncat(s2, s3, s2_len);
 			}
 
-          		(void) strncat(s1, s2, strlen(s2));
+			s1_len -= strlen(s2);
+			if (s1_len > 0)
+				(void) strncat(s1, s2, s1_len);
           		cpuid_start = cpuid_arr[j];
            		l = 1;
 		} else {
 	        	if (k == ncpus) {
         	    		(void) snprintf(s2, sizeof (s2), "%d-%d",
                 			cpuid_start, cpuid_start + l);
-         			(void) strncat(s1, s2, strlen(s2));
+				s1_len -= strlen(s2);
+				if (s1_len > 0)
+					(void) strncat(s1, s2, s1_len);
        			} else {
             			l++;
        			}
```

Reply:
managed to recover it: https://github.com/intel/numatop/compare/eab0ac5253c5843aa0f0ac36e2eec7612207711b...c1001fd926c24eae2d40729492e07270ce133b72.patch